Documentation/concepts/performance/index.rst

```diff
@@ -0,0 +1,330 @@
+command on the servers.
+
+.. code-block:: shell-session
+
+  $ cat packet-hosts.ini
+  [master]
+  136.144.55.223 ansible_python_interpreter=python3 ansible_user=root prv_ip=10.67.33.131 node_ip=10.33.33.10 master=knb-0
+  [nodes]
+  136.144.55.225 ansible_python_interpreter=python3 ansible_user=root prv_ip=10.67.33.133 node_ip=10.33.33.11
+  $ ansible -i packet-hosts.ini all -m shell -a 'uptime'
+  136.144.55.223 | CHANGED | rc=0 >>
+  09:31:43 up 33 min,  1 user,  load average: 0.00, 0.00, 0.00
+  136.144.55.225 | CHANGED | rc=0 >>
+    09:31:44 up 33 min,  1 user,  load average: 0.00, 0.00, 0.00
+
+
+Next, we use the ``packet-disbond.yaml`` playbook to configure the network
+interfaces of the machines. This will destroy the ``bond0`` interface and
+configure the first physical interface with the public and private IPs
+(``prv_ip``) and the second with the node IP (``node_ip``) that will be used
+for our evaluations (see `Packet documentation
+<https://www.packet.com/resources/guides/layer-2-configurations/>`_ and our
+scripts for more info).
+
+.. code-block:: shell-session
+
+  $ ansible-playbook -i packet-hosts.ini playbooks/packet-disbond.yaml
+
+
+.. note::
+
+    For hardware platforms other than Packet, users need to provide their own
+    inventory file (``packet-hosts.ini``) and follow the subsequent steps.
+
+
+Install Required Software
+-------------------------
+
+Install netperf (used for raw host-to-host measurements):
+
+.. code-block:: shell-session
+
+  $ ansible-playbook -i packet-hosts.ini playbooks/install-misc.yaml
+
+
+Install ``kubeadm`` and its dependencies:
+
+.. code-block:: shell-session
+
+  $ ansible-playbook -i packet-hosts.ini playbooks/install-kubeadm.yaml
+
+We use `kubenetbench <https://github.com/cilium/kubenetbench>`_ to execute the
+`netperf <https://hewlettpackard.github.io/netperf/>`_ benchmark in a Kubernetes
+environment.  While kubenetbench was developed by Cilium developers, it is not
+specific to Cilium and can be executed on top of any Kubernetes deployment. In
+this report we focus on pod-to-pod communication between different nodes. To
+install kubenetbench:
+
+.. code-block:: shell-session
+
+  $ ansible-playbook -i packet-hosts.ini playbooks/install-kubenetbench.yaml
+
+
+Cleanup
+-------
+
+When done with benchmarking, the allocated Packet resources can be released with:
+
+.. code-block:: shell-session
+
+  $ cd terraform && terraform destroy -var 'packet_token=API_TOKEN' -var 'packet_project_id=PROJECT_ID'
+
+
+----------
+Evaluation
+----------
+
+.. _tunneling_results:
+
+Tunneling
+---------
+
+Configure Cilium in tunneling (:ref:`arch_overlay`) mode:
+
+.. code-block:: shell-session
+
+  $ ansible-playbook -e mode=tunneling -i packet-hosts.ini playbooks/install-k8s-cilium.yaml
+  $ ansible-playbook -e conf=vxlan -i packet-hosts.ini playbooks/run-kubenetbench.yaml
+
+The first command configures Cilium to use tunneling (``-e mode=tunneling``),
```

Comment:
Do we need to add any additional commands in between the above to ensure that the machines are restarted & ready? Maybe it should be split into two code blocks to make sure that anyone trying to reproduce will stop to verify this first?

Reply:
Will leave this one to potentially address by either @pchaigno or @kkourt .

Reply:
@kkourt could you also follow-up to include the "higher is better" / "lower is better" wording into every plot?

Reply:
Why would we need to restart the machines? If we need to, we can do it as part of the playbook.

Reply:
Sorry, I meant to add this feedback for line 216 which talks about clearing out old state.

Reply:
Ah I see. Maybe we could just check that BPF programs are unloaded as part of the second playbook (`playbooks/run-rawnetperf.yaml`). That should just be a matter of running `bpftool net`. I'm guessing that would be our main concern, right?

Reply:
AFAIU, the ansible play will wait until the machine _is_ rebooted so state should be clean after that. There are two reasons for the reboot: bpf programs (for which we could call cilium cleanup and check bpf progs), but also changes on the MTU of the NICs.